### PR TITLE
Fix edge case for normalize

### PR DESCRIPTION
**Pull request by zhongjiajie**

### CODE REVIEW

File: src/spdx_matcher/__init__.py [hunk 1]
```diff
@@ -190,6 +190,10 @@ def normalize(license_text, remove_sections=REMOVE_FINGERPRINT):
     # To avoid the possibility of a non-match due to different spacing of words, line breaks,
     # or paragraphs.
     license_text = re.sub(r" +", " ", " ".join(license_text.split()))
+
+    # To avoid the possibility of a non-match due to missing space before or after specific symbol
+    # including period, comma, question mark, exclamation mark, colon, semicolon.
+    license_text = re.sub(r"\s+([.,!])", r"\1", license_text)
```
Comment (zhongjiajie) [inline src/spdx_matcher/__init__.py:196]: Some licenses text with space before the period, we should also normalize them

File: tests/test_spdx_matcher.py [hunk 2]
```diff
@@ -40,10 +41,12 @@
...
 """
 
-with open("tests/APACHE.txt", mode="rt", encoding="utf-8") as af:
+current_dir = Path(__file__).parent
+
+with open(f"{current_dir}/APACHE.txt", mode="rt", encoding="utf-8") as af:
     APACHE2 = af.read()
 
-with open("tests/CHALLENGING.txt", mode="rt", encoding="utf-8") as af:
+with open(f"{current_dir}/CHALLENGING.txt", mode="rt", encoding="utf-8") as af:
```
Comment (zhongjiajie) [inline tests/test_spdx_matcher.py:49]: To make test file can be run in ide or in tests directory by `python test_spdx_matcher.py`

File: tests/test_spdx_matcher.py [hunk 3]
```diff
@@ -88,5 +91,13 @@ def test_backtracking_challenging(self):
         self.assertTrue("GPL-1.0-only" in analysis["licenses"])
 
 
+class TestNormalize(unittest.TestCase):
```
Comment (zhongjiajie) [inline tests/test_spdx_matcher.py:94]: Add some test about the normalize

**Comment by zhongjiajie** [comment]
Today, I doubt the correctness of this PR, because it is completely dependent on the data of SPDX license. Should I directly correct the dataset of SPDX license if there is a problem with the matching?

**Comment by MikeMoore63** [comment]
The tests failed so won't merge anyway. I did see a large change in thedata did you use the command line toool to rebuild cahce? The data should nnot be updated as it just reflects what comes from SPDX. The library does allow you to mainatin your own data but that is outside scoppe of this.

**Comment by MikeMoore63** [comment]
I solved the test problems and optimise dyour fix and validated cache see that its v3.25.0 of spdx data release and checks running now should have a new version availble in pypi in about 15 minutes

**Comment by zhongjiajie** [comment]
> I did see a large change in thedata did you use the command line toool to rebuild cahce?

Yeah, I rebuild the data from spdx license json data

**Comment by zhongjiajie** [comment]
Thanks Mike 👍

**Pull request merged**